packages/scripts/src/components/front-end/payments/paypal/buttons.ts

```diff
@@ -0,0 +1,334 @@
+  }
+
+  // Initialize PayPal buttons
+  try {
+    // Load SDK first
+    await loadPayPalSDK(config.clientId, config.sandbox, config.currency || 'USD');
+
+    // Wait a bit for the SDK to be fully ready
+    await new Promise((resolve) => setTimeout(resolve, 100));
+
+    const p = (window as unknown as { paypal?: PayPalSDK }).paypal;
+    if (!p) {
+      logError('PayPal SDK is not available on window');
+      return;
+    }
+
+    p.Buttons({
+      style: {
+        layout: 'vertical',
+        color: 'blue',
+        shape: 'rect',
+        label: 'paypal',
+      },
+      createOrder: async () => {
+        // Create order
+
+        try {
+          // Serialize current form values so server can compute dynamic amount before order create
+          const fd = new FormData(form);
+          const values: Values = {};
+          fd.forEach((v, k) => {
+            (values as Record<string, unknown>)[k] = v as unknown as string | number | boolean | File;
+          });
+
+          const order = await createOrder(ENDPOINTS.orders, config.integration, values);
+          if (!order || !order.id) {
+            logError('Failed to create order or no order ID returned', order);
+            throw new Error('Failed to create PayPal order');
+          }
+          // Order created
+          hiddenInput.value = order.id;
+          return order.id;
+        } catch (error) {
+          logError('Error in createOrder:', error);
+          throw error;
+        }
+      },
+      onApprove: async (data: { orderID: string }) => {
+        // Order approved: immediately capture, but do NOT auto-submit
+        hiddenInput.value = String(data.orderID || '');
+        try {
+          const fd = new FormData(form);
+          const values: Values = {};
+          fd.forEach((v, k) => {
+            (values as Record<string, unknown>)[k] = v as unknown as string | number | boolean | File;
+          });
+          const captureResult = await captureOrder(ENDPOINTS.orders, config.integration, String(data.orderID || ''), {
+            values,
+          });
+          if (captureResult && captureResult.status === 'COMPLETED') {
+            showNotice(root, 'Payment completed with PayPal. Click Submit to finish.');
+          } else {
+            logError('Payment capture failed', captureResult);
+            showNotice(root, 'PayPal capture failed. Please try again.');
+          }
+        } catch (error) {
+          logError('Error during payment capture:', error);
+          showNotice(root, 'PayPal capture error. Please try again.');
+        }
+      },
+      onError: (err: unknown) => {
+        logError('Payment error:', err);
+      },
+      onCancel: (_data: unknown) => {
+        // Payment cancelled
+      },
+    }).render(root);
+  } catch (error) {
+    logError('Failed to initialize buttons:', error);
+  }
+}
+
+document.addEventListener('DOMContentLoaded', async () => {
+  const roots = document.querySelectorAll(Q.root);
```

Comment:
Use `HTMLElement` types when querying html elements, as this makes certain available which are not present in `Element` alone.

```suggestion
  const roots = document.querySelectorAll<HTMLElement>(Q.root);
```